### ISSUE TITLE
Hiding builder implementation does not work

### ISSUE THREAD

**Issue by alexwen** [immutables/immutables#220]
Testing with 2.1.3 we can no longer hide the builder implementation due to a mis-match in the visibility of the builder method on the immutable class. This can be seen with the example from the documentation:

```
@Value.Style(visibility = Value.Style.ImplementationVisibility.PACKAGE)
@Value.Immutable
public abstract class Point {
    @Value.Parameter public abstract double x();
    @Value.Parameter public abstract double y();

    public static Point of(double x, double y) {
        return ImmutablePoint.of(x, y);
    }

    public static Builder builder() {
        return ImmutablePoint.builder();
    }
    // Signatures of abstract methods should match to be
    // overridden by implementation builder
    public interface Builder {
        Builder x(double x);
        Builder y(double y);
        Point build();
    }
}
```
When attempting to compile this with 2.1.3:
```
Error:(140, 33) java: builder() in ImmutablePoint cannot override builder() in Point
  attempting to assign weaker access privileges; was public
```

Let me know if I can provide any more information. Thanks for the help.

**Comment by elucash** [comment]
Thank you for bringing this up, fix will be in next nearest release

**Comment by elucash** [comment]
Please, evaluate the fix available in 2.1.4.

**Comment by alexwen** [comment]
It appears to be working as expected now, thanks!
